From 95174317dcae63a0f26a3263c1ed9fa97540890f Mon Sep 17 00:00:00 2001
From: Hannu Nyman <hannu.nyman@iki.fi>
Date: Thu, 23 Mar 2017 16:53:17 +0200
Subject: [PATCH] mac80211: ath10k: allow failure getting board id via otp

ath10k tries to fetch board id via otp, but that fails for many chips
like QCA988x, QCA9984 etc. Recent commit cc189c0b7f removed the earlier
hack that had allowed QCA radios to work, as that hack was incompatible
with the new wifi chips being introduced to the source tree.

Restore functionality for the existing wifi chips by modifying the
return value of the 'board id via otp' function to a value that is recognised
as a harmless error, so that name evaluation continues by using the board file.

Patch originally suggested by Christian Lamparter in forum discussion.

Signed-off-by: Hannu Nyman <hannu.nyman@iki.fi>
---
 .../mac80211/patches/936-ath10k-fix-otp-failure-result.patch  | 11 +++++++++++
 1 file changed, 11 insertions(+)
 create mode 100644 package/kernel/mac80211/patches/936-ath10k-fix-otp-failure-result.patch

diff --git a/package/kernel/mac80211/patches/936-ath10k-fix-otp-failure-result.patch b/package/kernel/mac80211/patches/936-ath10k-fix-otp-failure-result.patch
new file mode 100644
index 0000000000..e28fa8fa60
--- /dev/null
+++ b/package/kernel/mac80211/patches/936-ath10k-fix-otp-failure-result.patch
@@ -0,0 +1,11 @@
+--- a/drivers/net/wireless/ath/ath10k/core.c
++++ b/drivers/net/wireless/ath/ath10k/core.c
+@@ -686,7 +686,7 @@
+ 	if (ret) {
+ 		ath10k_err(ar, "could not execute otp for board id check: %d\n",
+ 			   ret);
+-		return ret;
++		return -EOPNOTSUPP;
+ 	}
+ 
+ 	board_id = MS(result, ATH10K_BMI_BOARD_ID_FROM_OTP);
-- 
2.13.2

